### ISSUE TITLE
normalize_tuple() change.

### ISSUE THREAD

**Issue by CuiDonCai** [keras-team/keras#17608]
normalize_tuple() changed its behavior since version r2.6 [source code](https://github.com/keras-team/keras/blob/r2.11/keras/utils/conv_utils.py). It is called by the tf.keras.layers.Cropping1D() function.
case with tf 1.15
```
import tensorflow as tf
import numpy as np
input_shape = (2, 3, 2)
x = np.arange(np.prod(input_shape)).reshape(input_shape)
y = tf.keras.layers.Cropping1D(cropping=(-10,0))(x)
````

![image](https://user-images.githubusercontent.com/14941351/221460372-35178405-183e-4e0e-8380-e9af0f996ed8.png)

it's fine. But with tf2.11
```
import tensorflow as tf
import numpy as np
input_shape = (2, 3, 2)
x = np.arange(np.prod(input_shape)).reshape(input_shape)
y = tf.keras.layers.Cropping1D(cropping=(-1,0))(x)
````
there is a error.
![image](https://user-images.githubusercontent.com/14941351/221460460-879e1d47-833e-42ce-ac36-3139b26f03e3.png)

so how to use Cropping1D() in tensorflow 2.11 consistent with the behavior in tensorflow 1.15. Or is there another way to do it in tensorflow 2.11? Thanks so much for response.

**Comment by SuryanarayanaY** [comment]
Hi @CuiDonCai ,

With TF 2.11v there is change in validations in order to raise the error for the cropping values<0 which seems missing with Tf 1.15 version.Hence with Tf 2.11v it is not possible the behaviour of Tf 1.15 version

Please check the code where the validations introduced for values <0.
With the below code any cropping values < 0 (negative values) will raise value Error which is intended.

Please refer the source.
https://github.com/keras-team/keras/blob/e6784e4302c7b8cd116b74a784f4b78d60e83c26/keras/layers/reshaping/cropping1d.py#L66

https://github.com/keras-team/keras/blob/e6784e4302c7b8cd116b74a784f4b78d60e83c26/keras/utils/conv_utils.py#L101-L113

It is not recommended to use Tf1.15 versions as validations missing there.

**Comment by CuiDonCai** [comment]
Yes, Thank you @SuryanarayanaY.  Actually I'm  upgrading the code of a model from tf 1.15 to tf2.11.  And there's an alternative function for Cropping1D().  tf.slice() seem not  a good idea. I'm not familiar with tf. Thank you very much.

**Comment by CuiDonCai** [comment]
Yes, Thank you @SuryanarayanaY. Actually I'm upgrading the code of a model from tf 1.15 to tf2.11. And is there an alternative function for Cropping1D() ?  tf.slice() seem not a good idea. I'm not familiar with tf. Thank you very much.

**Comment by SuryanarayanaY** [comment]
Hi @CuiDonCai ,
You can use `cropping1d` with positive values with latest versions. Only limitation is Negative values for the argument `cropping` is not allowed.

**Comment by CuiDonCai** [comment]
the tensor shape (None, None, 64) bothers me! If I want to a slice shape of (None, first_20, 64), it's no problem. but I want to get a slice shape of (None, last_20, 64). it's simple in tf 1.15:
```
x.shape = (None, None, 64) 
y = tf.keras.layers.Cropping1D(cropping=(-20,0))(x)
````
But how to implement the feature with latest versions. And I don't want to rebuild model with tf2.11, is there a simple way I can get specific value of the `None` with axis 1? Thank you!

**Comment by CuiDonCai** [comment]
em, it's seem have same problem when I want to  keep  the beginning data. Anyway I seem must  konw the value `None`.

**Comment by SuryanarayanaY** [comment]
Hi @CuiDonCai ,
Please look at the args:
```
    Args:
      cropping: Int or tuple of int (length 2)
        How many units should be trimmed off at the beginning and end of
        the cropping dimension (axis 1).
        If a single int is provided, the same value will be used for both.
    Input shape:
      3D tensor with shape `(batch_size, axis_to_crop, features)`
```
You should know the `axis_to_crop` value in input_shape to use this API. Without this value it can't work.Also note that the the cropping axis to be >=0 and <(axis_to_crop/2). 

For example if you have input_shape of (2,10,2) i.e 2 nos of 2D arrays with 10 rows and 2 columns each.If you want to get 5th row to 7 th row data of each 2D array, then value for argument cropping to be a tuple like cropping=(4, 10-7)= (4,3).
Please refer the attached code snippet below and the [gist](https://colab.research.google.com/gist/SuryanarayanaY/bd98ccc8fc688cf590b97454df5a644f/17608.ipynb) also for reference.

```
input_shape = (2, 10, 2)
x = np.arange(np.prod(input_shape)).reshape(input_shape)
print(x)

y = tf.keras.layers.Cropping1D(cropping=(4,3))(x)
print(y)
```

**Comment by CuiDonCai** [comment]
ok, thank you !
